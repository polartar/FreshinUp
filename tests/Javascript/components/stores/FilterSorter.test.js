import { createLocalVue, shallowMount, mount } from '@vue/test-utils'
import Component from '~/components/stores/FilterSorter.vue'

describe('FilterSorter', () => {
  // Component instance "under test"
  let localVue
  describe('Snapshots', () => {
    test('defaults', () => {
      localVue = createLocalVue()
      const wrapper = mount(Component, {
        localVue: localVue
      })
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Methods', () => {
    beforeEach(() => {
      localVue = createLocalVue()
    })

    test('selectLocation function change filters', () => {
      const wrapper = shallowMount(Component)
      wrapper.vm.selectLocation({ uuid: 1 }, () => {})
      expect(wrapper.vm.location_uuid).toBe(1)
      wrapper.vm.selectLocation(null, () => {})
      expect(wrapper.vm.location_uuid).toBe('')
    })

    test('selectTag function change filters', () => {
      const wrapper = shallowMount(Component)
      wrapper.vm.selectTag({ uuid: 1 }, () => {})
      expect(wrapper.vm.tag).toBe(1)
      wrapper.vm.selectTag(null, () => {})
      expect(wrapper.vm.tag).toBe('')
    })

    test('clearFilters function clear filters', () => {
      const wrapper = shallowMount(Component)
      wrapper.vm.clearFilters({})
      expect(wrapper.vm.status).toBeNull()
      expect(wrapper.vm.tag).toBeNull()
      expect(wrapper.vm.location_uuid).toBeNull()
    })

    test('run function emitted runFilter', () => {
      const wrapper = shallowMount(Component)
      wrapper.setData({
        status: 2,
        tag: '2',
        location_uuid: '3'
      })
      wrapper.vm.run({})
      expect(wrapper.emitted().runFilter).toBeTruthy()
      const runParams = wrapper.emitted().runFilter[0][0]
      expect(runParams['status']).toEqual(2)
      expect(runParams['tag']).toEqual('2')
      expect(runParams['location_uuid']).toEqual('3')
    })
  })

  describe('Computed', () => {
    beforeEach(() => {
      localVue = createLocalVue()
    })
    test('filters', () => {
      const wrapper = shallowMount(Component)
      wrapper.setData({
        status: 2,
        tag: '2',
        location_uuid: '3'
      })
      expect(wrapper.vm.filters.status).toBe(2)
      expect(wrapper.vm.filters.tag).toBe('2')
      expect(wrapper.vm.filters.location_uuid).toBe('3')
    })
  })
})
