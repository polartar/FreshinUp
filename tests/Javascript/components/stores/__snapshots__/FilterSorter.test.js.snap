// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterSorter Snapshots defaults 1`] = `
<div
  class="filter-transparent v-card v-sheet theme--light transparent"
>
  <form
    class="v-form"
    novalidate="novalidate"
  >
    <div
      class="container filter-sorter-container fluid"
    >
       
      <div
        class="layout filter-sorter-main-layout row wrap"
      >
        <!---->
         
        <div
          class="flex grow"
        >
          <div
            class="v-input v-text-field v-text-field--single-line v-text-field--solo v-text-field--solo-flat v-text-field--enclosed v-text-field--placeholder theme--light"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-text-field__slot"
                >
                  <input
                    placeholder="Start typing to Search"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="flex shrink"
        >
          <div
            class="v-input filter-sorter-sortable-select v-text-field v-text-field--single-line v-text-field--solo v-text-field--solo-flat v-text-field--enclosed v-select theme--light"
            role="combobox"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-select__slot"
                >
                  <label
                    aria-hidden="true"
                    class="v-label theme--light"
                    style="left: 0px; position: absolute;"
                  >
                    Sort by
                  </label>
                  <div
                    class="v-select__selections"
                  >
                    <input
                      aria-label="Sort by"
                      aria-readonly="false"
                      autocomplete="on"
                      data-vv-name="sort"
                      readonly="readonly"
                      type="text"
                    />
                  </div>
                  <div
                    class="v-input__append-inner"
                  >
                    <div
                      class="v-input__icon v-input__icon--append"
                    >
                      <i
                        aria-hidden="true"
                        class="v-icon material-icons theme--light"
                      >
                        arrow_drop_down
                      </i>
                    </div>
                  </div>
                </div>
                <div
                  class="v-menu"
                >
                  <div
                    class="v-menu__content theme--light "
                    style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                  >
                    <div
                      class="v-select-list v-card theme--light"
                    >
                      <div
                        class="v-list theme--light"
                        role="list"
                      >
                        <div
                          role="listitem"
                        >
                          <div
                            class="v-list__tile theme--light"
                          >
                            <div
                              class="v-list__tile__content"
                            >
                              <div
                                class="v-list__tile__title"
                              >
                                No data available
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="flex shrink"
        >
          <button
            class="v-btn v-btn--large theme--light ma-0 "
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
            Less Filters
          
            </div>
          </button>
        </div>
         
      </div>
       
      <hr
        class="v-divider theme--light"
      />
       
      <div
        class="layout filter-sorter-expanded-layout align-center justify-center row fill-height"
      >
        <div
          class="flex"
        >
          <div
            class="v-card__text px-0 pb-0"
          >
            <div
              class="layout row justify-space-between"
            >
              <div
                class="flex"
              >
                <div
                  class="layout row justify-space-between align-center mb-2"
                >
                  <div
                    class="white--text"
                  >
                    <strong
                      class="no-wrap"
                    >
                      
              Statuses
            
                    </strong>
                  </div>
                   
                  <button
                    class="mr-0 v-btn v-btn--flat v-btn--icon theme--light white--text"
                    color="white"
                    type="button"
                  >
                    <div
                      class="v-btn__content"
                    >
                      <i
                        aria-hidden="true"
                        class="v-icon clear-button-icon fa fa fa fa-times theme--light"
                      />
                    </div>
                  </button>
                </div>
                 
                <div
                  class="v-input v-text-field v-text-field--single-line v-text-field--solo v-text-field--solo-flat v-text-field--enclosed v-text-field--placeholder v-select v-input--hide-details theme--light"
                  flat=""
                  hide-details=""
                  item-text="name"
                  item-value="id"
                  placeholder="Select Status"
                  role="combobox"
                  solo=""
                >
                  <div
                    class="v-input__control"
                  >
                    <div
                      class="v-input__slot"
                    >
                      <div
                        class="v-select__slot"
                      >
                        <div
                          class="v-select__selections"
                        >
                          <input
                            aria-readonly="false"
                            autocomplete="on"
                            placeholder="Select Status"
                            readonly="readonly"
                            type="text"
                          />
                        </div>
                        <div
                          class="v-input__append-inner"
                        >
                          <div
                            class="v-input__icon v-input__icon--append"
                          >
                            <i
                              aria-hidden="true"
                              class="v-icon material-icons theme--light"
                            >
                              arrow_drop_down
                            </i>
                          </div>
                        </div>
                      </div>
                      <div
                        class="v-menu"
                      >
                        <div
                          class="v-menu__content theme--light "
                          style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                        >
                          <div
                            class="v-select-list v-card theme--light"
                          >
                            <div
                              class="v-list theme--light"
                              role="list"
                            >
                              <div
                                role="listitem"
                              >
                                <a
                                  class="v-list__tile v-list__tile--link theme--light"
                                >
                                  <div
                                    class="v-list__tile__action"
                                  >
                                    <i
                                      aria-hidden="true"
                                      class="v-icon fa fa-minus-square theme--light primary--text"
                                    />
                                  </div>
                                   
                                  <div
                                    class="v-list__tile__content"
                                  >
                                    <div
                                      class="v-list__tile__title"
                                    >
                                      All Status
                                    </div>
                                  </div>
                                </a>
                              </div>
                               
                              <hr
                                class="mt-2 v-divider theme--light"
                              />
                              <div
                                role="listitem"
                              >
                                <div
                                  class="v-list__tile theme--light"
                                >
                                  <div
                                    class="v-list__tile__content"
                                  >
                                    <div
                                      class="v-list__tile__title"
                                    >
                                      No data available
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
               
              <div
                class="flex ml-2"
              >
                <div
                  class="layout row justify-space-between mb-2"
                >
                  <div
                    class="white--text"
                  >
                    <strong
                      class="no-wrap"
                    >
                      
              Supplier
            
                    </strong>
                  </div>
                   
                  <!---->
                </div>
                 
                <div
                  background-color="white"
                  class="v-menu mt-0 pt-0"
                  flat=""
                  height="48"
                  not-clearable=""
                  placeholder="Select"
                  results-id-key="uuid"
                  solo=""
                  term-param="filter[name]"
                  url="companies?filter[type_key]=supplier"
                >
                  <div
                    class="v-input v-text-field v-text-field--single-line v-text-field--solo v-text-field--solo-flat v-text-field--enclosed v-text-field--placeholder v-input--hide-details v-input--is-readonly theme--light"
                  >
                    <div
                      class="v-input__control"
                    >
                      <div
                        class="v-input__slot white"
                        style="height: 48px;"
                      >
                        <div
                          class="v-text-field__slot"
                        >
                          <input
                            not-clearable=""
                            placeholder="Select"
                            readonly="readonly"
                            results-id-key="uuid"
                            term-param="filter[name]"
                            type="text"
                            url="companies?filter[type_key]=supplier"
                          />
                        </div>
                        <div
                          class="v-input__append-inner"
                        >
                          <div
                            class="v-input__icon v-input__icon--append"
                          >
                            <i
                              aria-hidden="true"
                              class="v-icon fas fa-caret-down theme--light"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="v-menu__content theme--light "
                    style="max-height: auto; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                  >
                     
                    <div
                      autofocus="autofocus"
                      background-color="white"
                      class="v-input v-text-field v-text-field--single-line v-text-field--solo v-text-field--solo-flat v-text-field--enclosed v-text-field--placeholder v-select v-autocomplete v-input--hide-details theme--light"
                      flat=""
                      height="48"
                      menu-props="[object Object]"
                      solo=""
                    >
                      <div
                        class="v-input__control"
                      >
                        <div
                          class="v-input__slot white"
                          style="height: 48px;"
                        >
                          <div
                            class="v-select__slot"
                          >
                            <div
                              class="v-select__selections"
                            >
                              <input
                                aria-label=""
                                autocomplete="off"
                                autofocus="autofocus"
                                placeholder="Select"
                                role="combobox"
                                type="text"
                              />
                            </div>
                            <div
                              class="v-input__append-inner"
                            >
                              <div
                                class="v-input__icon v-input__icon--append"
                              >
                                <i
                                  aria-hidden="true"
                                  class="v-icon material-icons theme--light"
                                >
                                  arrow_drop_down
                                </i>
                              </div>
                            </div>
                          </div>
                          <div
                            class="v-menu"
                          >
                            <div
                              class="v-menu__content theme--light v-autocomplete__content"
                              style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                            >
                              <div
                                class="v-select-list v-card theme--light"
                              >
                                <div
                                  class="v-list theme--light"
                                  role="list"
                                >
                                  <div
                                    role="listitem"
                                    style="display: none;"
                                  >
                                    <div
                                      class="v-list__tile theme--light"
                                    >
                                      <span
                                        class="search-section-title"
                                      >
                                        SELECTION
                                      </span>
                                    </div>
                                  </div>
                                    
                                  <div
                                    role="listitem"
                                    style="display: none;"
                                  >
                                    <div
                                      class="v-list__tile theme--light"
                                    >
                                      <span
                                        class="search-section-title"
                                      >
                                        RESULTS
                                      </span>
                                    </div>
                                  </div>
                                  <div
                                    role="listitem"
                                  >
                                    <div
                                      class="v-list__tile theme--light"
                                    >
                                      <div
                                        class="v-list__tile__content"
                                      >
                                        <div
                                          class="v-list__tile__title"
                                        >
                                          No data available
                                        </div>
                                      </div>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
               
              <div
                class="flex ml-2"
              >
                <div
                  class="layout row justify-space-between mb-2"
                >
                  <div
                    class="white--text"
                  >
                    <strong
                      class="no-wrap"
                    >
                      
              Hometown
            
                    </strong>
                  </div>
                   
                  <!---->
                </div>
                 
                <div
                  background-color="white"
                  class="v-input mt-0 pt-0 v-text-field v-text-field--single-line v-text-field--solo v-text-field--solo-flat v-text-field--enclosed v-text-field--placeholder v-select v-autocomplete v-input--hide-details theme--light"
                  flat=""
                  height="48"
                  solo=""
                >
                  <div
                    class="v-input__control"
                  >
                    <div
                      class="v-input__slot white"
                      style="height: 48px;"
                    >
                      <div
                        class="v-select__slot"
                      >
                        <input
                          aria-label=""
                          autocomplete="off"
                          placeholder="Select"
                          role="combobox"
                          type="text"
                        />
                        <div
                          class="v-input__append-inner"
                        >
                          <div
                            class="v-input__icon v-input__icon--append"
                          >
                            <i
                              aria-hidden="true"
                              class="v-icon material-icons theme--light"
                            >
                              arrow_drop_down
                            </i>
                          </div>
                        </div>
                      </div>
                      <div
                        class="v-menu"
                      >
                        <div
                          class="v-menu__content theme--light v-autocomplete__content"
                          style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                        >
                          <div
                            class="v-select-list v-card theme--light"
                          >
                            <div
                              class="v-list theme--light"
                              role="list"
                            >
                              <div
                                role="listitem"
                              >
                                <div
                                  class="v-list__tile theme--light"
                                >
                                  <div
                                    class="v-list__tile__content"
                                  >
                                    <div
                                      class="v-list__tile__title"
                                    >
                                      No data available
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
               
              <div
                class="flex ml-2"
              >
                <div
                  class="layout row justify-space-between mb-2"
                >
                  <div
                    class="white--text"
                  >
                    <strong
                      class="no-wrap"
                    >
                      
              Tags
            
                    </strong>
                  </div>
                   
                  <!---->
                </div>
                 
                <div
                  background-color="white"
                  class="v-menu mt-0 pt-0"
                  flat=""
                  height="48"
                  not-clearable=""
                  placeholder="Search Tag"
                  results-id-key="uuid"
                  solo=""
                  term-param="filter[name]"
                  url="foodfleet/store-tags"
                >
                  <div
                    class="v-input v-text-field v-text-field--single-line v-text-field--solo v-text-field--solo-flat v-text-field--enclosed v-text-field--placeholder v-input--hide-details v-input--is-readonly theme--light"
                  >
                    <div
                      class="v-input__control"
                    >
                      <div
                        class="v-input__slot white"
                        style="height: 48px;"
                      >
                        <div
                          class="v-text-field__slot"
                        >
                          <input
                            not-clearable=""
                            placeholder="Search Tag"
                            readonly="readonly"
                            results-id-key="uuid"
                            term-param="filter[name]"
                            type="text"
                            url="foodfleet/store-tags"
                          />
                        </div>
                        <div
                          class="v-input__append-inner"
                        >
                          <div
                            class="v-input__icon v-input__icon--append"
                          >
                            <i
                              aria-hidden="true"
                              class="v-icon fas fa-caret-down theme--light"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="v-menu__content theme--light "
                    style="max-height: auto; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                  >
                     
                    <div
                      autofocus="autofocus"
                      background-color="white"
                      class="v-input v-text-field v-text-field--single-line v-text-field--solo v-text-field--solo-flat v-text-field--enclosed v-text-field--placeholder v-select v-autocomplete v-input--hide-details theme--light"
                      flat=""
                      height="48"
                      menu-props="[object Object]"
                      solo=""
                    >
                      <div
                        class="v-input__control"
                      >
                        <div
                          class="v-input__slot white"
                          style="height: 48px;"
                        >
                          <div
                            class="v-select__slot"
                          >
                            <div
                              class="v-select__selections"
                            >
                              <input
                                aria-label=""
                                autocomplete="off"
                                autofocus="autofocus"
                                placeholder="Search Tag"
                                role="combobox"
                                type="text"
                              />
                            </div>
                            <div
                              class="v-input__append-inner"
                            >
                              <div
                                class="v-input__icon v-input__icon--append"
                              >
                                <i
                                  aria-hidden="true"
                                  class="v-icon material-icons theme--light"
                                >
                                  arrow_drop_down
                                </i>
                              </div>
                            </div>
                          </div>
                          <div
                            class="v-menu"
                          >
                            <div
                              class="v-menu__content theme--light v-autocomplete__content"
                              style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                            >
                              <div
                                class="v-select-list v-card theme--light"
                              >
                                <div
                                  class="v-list theme--light"
                                  role="list"
                                >
                                  <div
                                    role="listitem"
                                    style="display: none;"
                                  >
                                    <div
                                      class="v-list__tile theme--light"
                                    >
                                      <span
                                        class="search-section-title"
                                      >
                                        SELECTION
                                      </span>
                                    </div>
                                  </div>
                                    
                                  <div
                                    role="listitem"
                                    style="display: none;"
                                  >
                                    <div
                                      class="v-list__tile theme--light"
                                    >
                                      <span
                                        class="search-section-title"
                                      >
                                        RESULTS
                                      </span>
                                    </div>
                                  </div>
                                  <div
                                    role="listitem"
                                  >
                                    <div
                                      class="v-list__tile theme--light"
                                    >
                                      <div
                                        class="v-list__tile__content"
                                      >
                                        <div
                                          class="v-list__tile__title"
                                        >
                                          No data available
                                        </div>
                                      </div>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="flex text-no-wrap"
        >
          <button
            class="v-btn v-btn--large theme--light"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
            Clear Filters
          
            </div>
          </button>
        </div>
      </div>
    </div>
  </form>
</div>
`;
